### PR TITLE
Add HP and Lenovo Firmware

### PR DESCRIPTION
**Pull request by stefdb**

### CODE REVIEW

File: WHENCE [hunk 1]
```diff
@@ -5926,6 +5926,119 @@ Driver: cs35l41_hda - CS35l41 ALSA HDA audio driver
 
 File: cirrus/cs35l41-dsp1-spk-prot.wmfw
 File: cirrus/cs35l41-dsp1-spk-prot.bin
+File: cirrus/cs35l41/v6.61.1/halo_cspl_RAM_revB2_29.63.1.wmfw
+File: cirrus/cs35l41/v6.61.1/halo_cspl_RAM_diag_revB2_29.63.1.wmfw
+Link: cirrus/cs35l41-dsp1-spk-prot-103c8971.wmfw -> cs35l41/v6.61.1/halo_cspl_RAM_revB2_29.63.1.wmfw
+Link: cirrus/cs35l41-dsp1-spk-cali-103c8971.wmfw -> cs35l41/v6.61.1/halo_cspl_RAM_revB2_29.63.1.wmfw
+Link: cirrus/cs35l41-dsp1-spk-diag-103c8971.wmfw -> cs35l41/v6.61.1/halo_cspl_RAM_diag_revB2_29.63.1.wmfw
```
Comment (stuhenderson) [inline WHENCE]: shall we drop diag to make this a bit more palatable?

Comment (stefdb) [reply WHENCE]: Done

**Review by stuhenderson — APPROVED** [review]

**Review by stuhenderson — CHANGES_REQUESTED** [review]
There's some "Empty files" in the binaries.  should these be there?

**Comment by stefdb** [comment]
I've fixed the empty files now. This raises an interesting question. We dont have all laptops that these firmwares are for, so we cant test it. Do we want to talk to HP/Lenovo before we submit them so they can test them?

**Review by stuhenderson — APPROVED** [review]

**Pull request merged**